### DISCUSSION TITLE
No data received from Bosch Twinguard 8750001213

### DISCUSSION THREAD

**Discussion by Sailing74** [Koenkk/zigbee2mqtt#16993] [Devices]
I was able to pair a Bosch Twinguard 8750001213 by pairing it just with the coordinator (doesn't work with routers apparently), and only by submitting the install code via the webinterface (doesn't work via mqtt apparently). This device appears to be much more picky than other current Bosch zigbee devices.

However, after pairing, no data is received from the device. Hours after the successul pairs, all values are still "N/A" - with the exception of link quality, that is shown.

Pairing log:

`info  2023-03-10 21:08:05: Removing device '0x000d6f0016b30764' (block: false, force: true)
info  2023-03-10 21:08:05: MQTT publish: topic 'zigbee2mqtt/0x000d6f0016b30764', payload ''
info  2023-03-10 21:08:05: Successfully removed device '0x000d6f0016b30764' (block: false, force: true)
info  2023-03-10 21:08:06: MQTT publish: topic 'zigbee2mqtt/bridge/response/device/remove', payload '{"data":{"block":false,"force":true,"id":"0x000d6f0016b30764"},"status":"ok","transaction":"k7v3u-8"}'
info  2023-03-10 21:08:35: Zigbee: disabling joining new devices.
info  2023-03-10 21:08:35: MQTT publish: topic 'zigbee2mqtt/bridge/response/permit_join', payload '{"data":{"device":"Coordinator","time":254,"value":false},"status":"ok","transaction":"k7v3u-9"}'
info  2023-03-10 21:08:38: Zigbee: allowing new devices to join via Coordinator.
info  2023-03-10 21:08:38: MQTT publish: topic 'zigbee2mqtt/bridge/response/permit_join', payload '{"data":{"device":"Coordinator","time":254,"value":true},"status":"ok","transaction":"k7v3u-10"}'
info  2023-03-10 21:09:16: Device '0x000d6f0016b30764' joined
info  2023-03-10 21:09:16: MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"0x000d6f0016b30764","ieee_address":"0x000d6f0016b30764"},"type":"device_joined"}'
info  2023-03-10 21:09:16: Starting interview of '0x000d6f0016b30764'
info  2023-03-10 21:09:17: MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"0x000d6f0016b30764","ieee_address":"0x000d6f0016b30764","status":"started"},"type":"device_interview"}'
info  2023-03-10 21:09:17: MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"0x000d6f0016b30764","ieee_address":"0x000d6f0016b30764"},"type":"device_announce"}'
error 2023-03-10 21:09:37: Failed to interview '0x000d6f0016b30764', device has not successfully been paired
info  2023-03-10 21:09:37: MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"0x000d6f0016b30764","ieee_address":"0x000d6f0016b30764","status":"failed"},"type":"device_interview"}'
info  2023-03-10 21:10:51: MQTT publish: topic 'zigbee2mqtt/bridge/response/device/configure', payload '{"data":{"id":"0x000d6f0016b30764"},"status":"ok","transaction":"k7v3u-11"}'
info  2023-03-11 00:16:18: Successfully added new install code
info  2023-03-11 00:16:18: MQTT publish: topic 'zigbee2mqtt/bridge/response/install_code/add', payload '{"data":{"value":"RB01SGXXX"},"status":"ok","transaction":"k7v3u-12"}'
info  2023-03-11 00:16:18: Zigbee: allowing new devices to join.
info  2023-03-11 00:16:18: MQTT publish: topic 'zigbee2mqtt/bridge/response/permit_join', payload '{"data":{"time":254,"value":true},"status":"ok","transaction":"k7v3u-13"}'
info  2023-03-11 00:16:24: Starting interview of '0x000d6f0016b30764'
info  2023-03-11 00:16:25: MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"0x000d6f0016b30764","ieee_address":"0x000d6f0016b30764","status":"started"},"type":"device_interview"}'
info  2023-03-11 00:16:25: MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"0x000d6f0016b30764","ieee_address":"0x000d6f0016b30764"},"type":"device_announce"}'
info  2023-03-11 00:17:09: Successfully interviewed '0x000d6f0016b30764', device has successfully been paired
info  2023-03-11 00:17:09: Device '0x000d6f0016b30764' is supported, identified as: Bosch Twinguard (8750001213)
info  2023-03-11 00:17:09: MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"definition":{"description":"Twinguard","exposes":[{"access":1,"description":"Indicates whether the device detected smoke","name":"smoke","property":"smoke","type":"binary","value_off":false,"value_on":true},{"access":1,"description":"Measured temperature value","name":"temperature","property":"temperature","type":"numeric","unit":"°C"},{"access":1,"description":"Measured relative humidity","name":"humidity","property":"humidity","type":"numeric","unit":"%"},{"access":1,"description":"The measured CO2 (carbon dioxide) value","name":"co2","property":"co2","type":"numeric","unit":"ppm"},{"access":1,"description":"Measured illuminance in lux","name":"illuminance_lux","property":"illuminance_lux","type":"numeric","unit":"lx"},{"access":1,"description":"Remaining battery in %, can take up to 24 hours before reported.","name":"battery","property":"battery","type":"numeric","unit":"%","value_max":100,"value_min":0},{"access":7,"description":"Mode of the alarm (sound effect)","name":"alarm","property":"alarm","type":"enum","values":["stop","pre_alarm","fire","burglar"]},{"access":1,"description":"Siren state","name":"siren_state","property":"siren_state","type":"text"},{"access":7,"description":"Initiate self-test","name":"self_test","property":"self_test","type":"binary","value_off":false,"value_on":true},{"access":7,"description":"Sensitivity of the smoke alarm","name":"sensitivity","property":"sensitivity","type":"enum","values":["low","medium","high"]},{"access":7,"description":"Enable/disable pre-alarm","name":"pre_alarm","property":"pre_alarm","type":"enum","values":["OFF","ON"]},{"access":7,"description":"Enable/disable heartbeat","name":"heartbeat","property":"heartbeat","type":"enum","values":["OFF","ON"]},{"access":1,"description":"Link quality (signal strength)","name":"linkquality","property":"linkquality","type":"numeric","unit":"lqi","value_max":255,"value_min":0}],"model":"8750001213","options":[{"access":2,"description":"Number of digits after decimal point for temperature, takes into effect on next report of device.","name":"temperature_precision","property":"temperature_precision","type":"numeric","value_max":3,"value_min":0},{"access":2,"description":"Calibrates the temperature value (absolute offset), takes into effect on next report of device.","name":"temperature_calibration","property":"temperature_calibration","type":"numeric"},{"access":2,"description":"Number of digits after decimal point for humidity, takes into effect on next report of device.","name":"humidity_precision","property":"humidity_precision","type":"numeric","value_max":3,"value_min":0},{"access":2,"description":"Calibrates the humidity value (absolute offset), takes into effect on next report of device.","name":"humidity_calibration","property":"humidity_calibration","type":"numeric"},{"access":2,"description":"Calibrates the illuminance_lux value (percentual offset), takes into effect on next report of device.","name":"illuminance_lux_calibration","property":"illuminance_lux_calibration","type":"numeric"}],"supports_ota":false,"vendor":"Bosch"},"friendly_name":"0x000d6f0016b30764","ieee_address":"0x000d6f0016b30764","status":"successful","supported":true},"type":"device_interview"}'`

Anything I need to do in order to get some data from this device?

**Comment by Sailing74** [comment]
Ok, not sure what the issue is:

- bought a used Bosch controller. The Twinguard pairs and send data there.
- Paired again with z2m. z2m says "successfully interviewedl, but the Twinguard keeps blinking yellow and then, after some time, lights up red, indicating that pairing has failed.
- Next pairing attempt, remove and reinsert batteries; not specifically only setting the controller to pair, z2m says pairing has failed, but the Twinguard blinks green (thinks that pairing has succeeded). I restart z2m, and for some reason z2m has now forgotten that the interview failed, also thinking that pairing did succeed...?

I can trigger test alarms and the device has now started to transmit data.

For what it's worth, it appears that when pairing a Twinguard, one should keep trying until the Twinguard lights up green, indicating that pairing has succeeded, and then restart z2m. All this assumes that you have a controller with which Bosch install codes work.

**Reply by Chartreusito** [reply]
Did you end up using the device ?
I just landed a 8750001215 by vendor mistake and was able to interview the device successfully using the [pairing](https://www.zigbee2mqtt.io/devices/8750001213.html#pairing) procedure but i don't get any other data.

**Reply by Sailing74** [reply]
For me, the procedure described in the last paragraph of my above comment (pair until the device acknowledges the pairing by blinking green, then restart z2m) has meanwhile worked for three of these devices.

**Reply by MelindaJaphines** [reply]
> Did you end up using the device ? I just landed a 8750001215 by vendor mistake and was able to interview the device successfully using the [pairing](https://www.zigbee2mqtt.io/devices/8750001213.html#pairing) procedure but i don't get any other data.

Did you ever find a solution to this? I'm experiencing the same with model 8750001215. Managed to pair but no data.

**Reply by sarangnemo** [reply]
I have model 8750001215. For those interested, I just made a [PR](https://github.com/zigpy/zha-device-handlers/pull/3293) for a ZHA quirk which gets the measurements to update.

For me, HA (via ZHA) will report successful pairing using the 'permit' with QR code procedure, but the Twinguard also keeps blinking and eventually glows red. But if I just hold the Twinguard's button to put it back in pairing mode, it'll then glow green after a few seconds. (It never works if I do a full reset with removing batteries etc.) It seems likely that the measurement updates may only come through if you get the device to glow green upon pairing...
